coffee/shepherd.coffee

```diff
@@ -117,6 +117,13 @@ class Step extends Evented
     @tether = new Tether extend(tetherOpts, @options.tetherOptions)
 
   show: =>
+    if @options && @options.beforeShowPromise && typeof @options.beforeShowPromise is 'function'
+      @options.beforeShowPromise().then(=>
```

Comment:
In this scenario, I'm probably more likely to do `.then(=> @_show())` on one line since spreading it out doesn't really do much besides making it harder to read. Since we don't have any args to pass along to `@_show` and `@_show` is defined with a fat arrow (`=>`), we can do `.then(@_show)`.